v2/lint/base.go

```diff
@@ -83,14 +83,19 @@ func (l *Lint) CheckEffective(c *x509.Certificate) bool {
 // CheckEffective()
 // Execute()
 func (l *Lint) Execute(cert *x509.Certificate) *LintResult {
-	if l.Source == CABFBaselineRequirements && !util.IsServerAuthCert(cert) {
+	if l.Source == CABFBaselineRequirements && !util.IsServerAuthCert(cert) && !util.IsDelegatedOCSPResponderCert(cert) {
+		// OCSP certificates also fall under certain BRG lints, even if they don't have the ServerAuth EKU set
 		return &LintResult{Status: NA}
 	}
+
 	if !l.Lint.CheckApplies(cert) {
 		return &LintResult{Status: NA}
-	} else if !l.CheckEffective(cert) {
-		return &LintResult{Status: NE}
+	} else {
+		if !l.CheckEffective(cert) {
+			return &LintResult{Status: NE}
+		}
```

Comment:
I think, with the new edits, you can restore this to what it was, right? This rewrite has the same processing as the original code, and the original code is more concise, so we'd prefer that as the idiom. Unless I'm overlooking an edge case here?

Reply:
fixed.